analysis_options.yaml

```diff
@@ -15,69 +15,130 @@ linter:
     - always_put_control_body_on_new_line
     - always_require_non_null_named_parameters
     - annotate_overrides
+    - avoid_double_and_int_checks
+    - avoid_dynamic_calls
     - avoid_empty_else
+    - avoid_equals_and_hash_code_on_mutable_classes
+    - avoid_escaping_inner_quotes
     - avoid_field_initializers_in_const_classes
     - avoid_init_to_null
+    - avoid_js_rounded_ints
     - avoid_null_checks_in_equality_operators
+    - avoid_private_typedef_functions
+    - avoid_redundant_argument_values
     - avoid_relative_lib_imports
     - avoid_return_types_on_setters
+    - avoid_shadowing_type_parameters
     - avoid_slow_async_io
+    - avoid_type_to_string
+    - avoid_types_as_parameter_names
+    - avoid_unused_constructor_parameters
     - await_only_futures
+    - camel_case_extensions
     - camel_case_types
     - cancel_subscriptions
+    - cast_nullable_to_non_nullable
+    - close_sinks
+    - comment_references
+    - constant_identifier_names
     - control_flow_in_finally
+    - curly_braces_in_flow_control_structures
+    - directives_ordering
+    - do_not_use_environment
     - empty_catches
     - empty_constructor_bodies
     - empty_statements
+    - exhaustive_cases
+    - file_names
     - hash_and_equals
     - implementation_imports
+    - invariant_booleans
     - iterable_contains_unrelated_type
+    - join_return_with_assignment
     - library_names
     - library_prefixes
     - list_remove_unrelated_type
+    - literal_only_boolean_expressions
+    - missing_whitespace_between_adjacent_strings
     - no_adjacent_strings_in_list
     - no_duplicate_case_values
-    - non_constant_identifier_names
+    - no_runtimeType_toString
+    - omit_local_variable_types
```

Comment:
Is this rule smart enough to detect cases where we need an empty variable? Because on that case we need to add the type otherwise the variable will be dynamic.

Reply:
IMO declaring the type is always better than `var` but that may be trauma from JS.

Reply:
haha var in dart infers, unlike JS. same thing we already do for final. I think if it's not initialized then you are forced to use the type, but I will test rn

Reply:
yes if you don't initialize you are forced to type:

```
Prefer typing uninitialized variables and fields.dart(prefer_typing_uninitialized_variables)
```

so:
```
  var a = 1.0; // good
  double b; // good
  var c; // bad (would be dynamic)
  double d = 1.0; // bad (redundant)
```

Reply:
Sounds good then! I don't like car as well, but it seems smart enough.

Reply:
We all know you only like motorbikes! :p 

Jokes aside, I don't like mutable variables at all. But they are a necessary evil. And the keyword var can bring terrible JS memories but kt/dark pretty much erased that dark past for me.

Reply:
hahahha, I don't remember what I wanted to write there, I do like cars, but I do prefer bikes 🏍️